File: examples/Java/rrworker.java
package guide;

import org.zeromq.SocketType;
import org.zeromq.ZMQ;
import org.zeromq.ZMQ.Socket;
import org.zeromq.ZContext;

//  Hello World worker
//  Connects REP socket to tcp://*:5560
//  Expects "Hello" from client, replies with "World"
public class rrworker
{
    public static void main(String[] args) throws Exception
    {
        try (ZContext context = new ZContext()) {
            //  Socket to talk to server
            Socket responder = context.createSocket(SocketType.REP);
            responder.connect("tcp://localhost:5560");

            while (!Thread.currentThread().isInterrupted()) {
                //  Wait for next request from client
                String string = responder.recvStr(0);
                System.out.printf("Received request: [%s]\n", string);

                //  Do some 'work'
                Thread.sleep(1000);

                //  Send reply back to client
                responder.send("World");
            }
        }
    }
}

File: examples/Clojure/rrworker.clj
;; Hello World worker
;; Connects REP socket to tcp://*:5560
;; Expects "Hello" from client, replies with "World"

(ns rrworker
  (:require [zeromq.zmq :as zmq]))

(defn -main []
  (let [context (zmq/zcontext)]
    (with-open [responder (doto (zmq/socket context :rep)
                            (zmq/connect "tcp://127.0.0.1:5560"))]
      (while (not (.. Thread currentThread isInterrupted))
        (let [string (zmq/receive-str responder)]
          (printf "Received request: [%s]\n", string)
          ;; Do some work
          (Thread/sleep 1000)
          (zmq/send-str responder "World"))))))
